swagger.yaml

```diff
@@ -1198,6 +1540,119 @@ components:
           type: boolean
           description: Boolean value to indicate if option is correct or incorrect
 
+    VacationType:
+      type: string
+      description: Enum for vacation type
+      enum:
+        - VACATION
+        - UNPAID_TIME_OFF
+        - SICKNESS
+        - PERSONAL_DAYS
+        - MATERNITY_PATERNITY
+        - CHILD_SICKNESS
+
+    VacationRequestStatuses:
+      type: string
+      description: Enum for vacation request status
+      enum:
+        - PENDING
+        - APPROVED
+        - DECLINED
+
+    VacationRequest:
+      type: object
+      description: Person's single vacation request
+      required:
+        - personId
+        - days
+        - startDate
+        - endDate
+        - type
+        - message
+        - createdAt
+        - updatedAt
+        - draft
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: ID of vacation request in UUID
+          default: null
+        personId:
+          type: string
+          format: uuid
+          description: Person keycloak ID in UUID
+        draft:
+          type: boolean
+          description: Draft state
+        startDate:
+          type: string
+          format: date
+          description: Date of vacation start
+        endDate:
+          type: string
+          format: date
+          description: Date of vacation end
+        days:
+          type: integer
+          format: int32
+          description: Vacation days used for requested vacation
+        type:
+          $ref: "#/components/schemas/VacationType"
+        message:
+          type: string
+          description: Message attached to request
+        createdBy:
+          type: string
+          format: uuid
+          description: ID of person who created status in UUID
```

Comment:
Person's Keycloak ID ?

Reply:
yeah, createdBy: / type: string / format: uuid / description: Person keycloak ID in UUID